apps/api/src/domain/models/categories/entities/category.entity.ts

```diff
@@ -20,4 +28,37 @@ export class Category extends BaseDomainModel {
 
         this.members = cloneToPlainObject(members);
     }
+
+    validateExternalState({
+        resources,
+        connections,
+    }: DeepPartial<InMemorySnapshot>): Valid | InternalError {
+        const connectionCompositeIdsInSnapshot = (connections || []).map(({ id }) => ({
+            type: noteType,
+            id,
+        }));
+
+        const resourceCompositeIdentifiersInSnapshot = Object.values(resources || []).flatMap(
+            (resourceInstances) =>
+                resourceInstances.map((instance) => instance.getCompositeIdentifier())
+        );
+
+        const compositeIdentifiersInSnapshot = [
+            ...resourceCompositeIdentifiersInSnapshot,
+            ...connectionCompositeIdsInSnapshot,
+        ];
+
+        // TODO [performance] Optimize this if performance becomes an issue
+        const missingCompositeIDs = this.members.filter(
```

Comment:
Can we modify this so it's not nested?

Reply:
I'm thinking you mean take way the nested loop? Let's wait until we hit performance issues then optimize the algorithm.